cmd/tailscale/main.go

```diff
@@ -1292,7 +1310,10 @@ func (a *App) processUIEvents(w *app.Window, events []UIEvent, act jni.Object, s
 			requestBackend(e)
 		case ExitAllowLANEvent:
 			requestBackend(e)
-		case WebAuthEvent:
+		case WebAuthAndMaybeSetServerEvent:
+			if srv, _ := a.store.ReadString(customLoginServerPrefKey, ""); srv != state.backend.Prefs.ControlURL {
+				e.URL = srv
```

Comment:
I see. It's a control plane URL but only set if it differs from what we had before.